src/QemuRunner.jl

```diff
@@ -0,0 +1,170 @@
+our crossbuild environment.  Use `run()` to actually run commands within the
+`QemuRunner`, and `runshell()` as a quick way to get an interactive shell
+within the crossbuild environment.
+"""
+type QemuRunner <: Runner
+    qemu_cmd::Cmd
+    append_line::String
+    sandbox_cmd::Cmd
+    platform::Platform
+end
+
+qemu_path() = "/Users/kfischer/Projects/qemu-hvf/x86_64-softmmu/qemu-system-x86_64"
+kernel_path() = "/Users/kfischer/Projects/qemu-hvf/x86_64-softmmu/bzImage"
+
+function platform_def_mounts(platform)
+    tp = triplet(platform)
+    mapping = Pair{String,String}[
+        shard_path_squashfs(tp) => joinpath("/opt", tp)
+    ]
+
+    # We might also need the x86_64-linux-gnu platform for bootstrapping,
+    # so make sure that's always included
+    if platform != Linux(:x86_64)
+        ltp = triplet(Linux(:x86_64))
+        push!(mapping, shard_path_squashfs(ltp) => joinpath("/opt", ltp))
+    end
+
+    # If we're trying to run macOS and we have an SDK directory, mount that!
+    #=
+    if platform == MacOS()
+        sdk_version = "MacOSX10.10.sdk"
+        sdk_shard_path = joinpath(shards_cache, sdk_version)
+        push!(mapping, sdk_shard_path => joinpath("/opt", tp, sdk_version))
+    end
+    =#
+
+    # Reverse mapping order, because `sandbox` reads them backwards
+    reverse!(mapping)
+    return mapping
+end
+
+function QemuRunner(workspace_root::String; cwd = nothing,
+                      platform::Platform = platform_key(),
+                      extra_env=Dict{String, String}(),
+                      verbose::Bool = true,
+                      mounts = platform_def_mounts(platform))
+    global sandbox_path
+
+    # Ensure the rootfs for this platform is downloaded and up to date
+    update_rootfs(
+        platform != Linux(:x86_64) ?
+        [triplet(platform), triplet(Linux(:x86_64))] :
+        triplet(platform); verbose=verbose, squashfs=true)
+
+    qemu_cmd = ```
+        $(qemu_path()) -kernel $(kernel_path())
+        -M accel=hvf -nographic
+        -drive if=virtio,file=$(rootfs_path_squashfs()),format=raw
+        -nodefaults -rtc base=utc,driftfix=slew
+        -device virtio-serial -chardev stdio,id=charconsole0
+        -device virtconsole,chardev=charconsole0,id=console0
+        -fsdev local,security_model=none,id=fsdev0,path=$workspace_root -device virtio-9p-pci,id=fs0,fsdev=fsdev0,mount_tag=workspace
+    ```
+    append_line = "quiet console=hvc0 root=/dev/vda rootflags=ro rootfstype=squashfs noinitrd init=/sandbox"
+
+    sandbox_cmd = `--workspace 9p:workspace`
+    if cwd != nothing
+        sandbox_cmd = `$sandbox_cmd --cd $cwd`
+    end
+
+    c = 'b'
+    for (mount,dir) in mounts
+        qemu_cmd = `$qemu_cmd -drive if=virtio,file=$mount,format=raw`
+        sandbox_cmd = `$sandbox_cmd --map /dev/vd$c:$dir`
+        c = Char(Int(c) + 1)
+    end
+
+    sandbox_cmd = setenv(sandbox_cmd, merge(target_envs(triplet(platform)), extra_env))
```

Comment:
See my recent modifications to `UserNSRunner`; this `setenv()` is more or less useless.